core/common/src/main/java/alluxio/heartbeat/HeartbeatScheduler.java

```diff
@@ -90,7 +103,6 @@ public static void schedule(String threadName) {
         throw new RuntimeException("Timer for thread " + threadName + " not found.");
       }
       timer.schedule();
-      sTimers.remove(threadName);
```

Comment:
I move this to the end of the `timer.schedule()` on the line above it. I think it makes mores sense there so that all timer adding/removing is done in `ScheduledTimer`.